diff --git a/lib/server/siteMetadata.js b/lib/server/siteMetadata.js
index a8734a8291bdf7550b965d46bea37f226f18b0eb..c9608ce032e39625d33e52094784cce3139a37b3 100644
--- a/lib/server/siteMetadata.js
+++ b/lib/server/siteMetadata.js
@@ -10,19 +10,25 @@ exports.loadSiteVersion = loadSiteVersion;
 exports.loadPluginVersion = loadPluginVersion;
 exports.createSiteMetadata = createSiteMetadata;
 const tslib_1 = require("tslib");
+const fs = require('fs/promises');
 const fs_extra_1 = tslib_1.__importDefault(require("fs-extra"));
 const path_1 = tslib_1.__importDefault(require("path"));
 const utils_1 = require("@docusaurus/utils");
+
+async function loadPackageJson(packageJsonPath) {
+  return JSON.parse(await fs.readFile(packageJsonPath, 'utf8'));
+}
+
 async function loadPackageJsonVersion(packageJsonPath) {
     if (await fs_extra_1.default.pathExists(packageJsonPath)) {
         // eslint-disable-next-line @typescript-eslint/no-var-requires, import/no-dynamic-require, global-require
-        return require(packageJsonPath).version;
+        return (await loadPackageJson(packageJsonPath)).version;
     }
     return undefined;
 }
 async function loadPackageJsonName(packageJsonPath) {
     // eslint-disable-next-line @typescript-eslint/no-var-requires, import/no-dynamic-require, global-require
-    return require(packageJsonPath).name;
+    return (await loadPackageJson(packageJsonPath)).name;
 }
 async function loadSiteVersion(siteDir) {
     return loadPackageJsonVersion(path_1.default.join(siteDir, 'package.json'));
